lib/safemode.rb

```diff
@@ -39,20 +39,18 @@ def find_jail_class(klass)
   define_core_jail_classes
   
   class Box
+    attr_reader :output
     def initialize(delegate = nil, delegate_methods = [], filename = nil, line = nil)
-      @scope = Scope.new(delegate, delegate_methods)
+      @delegate = delegate
+      @delegate_methods = delegate_methods
       @filename = filename
       @line = line
     end    
 
     def eval(code, assigns = {}, locals = {}, &block)
       code = Parser.jail(code)
-      binding = @scope.bind(assigns, locals, &block)
-      result = Kernel.eval(code, binding, @filename || __FILE__, @line || __LINE__)
+      scope = Scope.new(@delegate, @delegate_methods, instance_vars: assigns, locals: locals, &block)
+      @output = Kernel.eval(code, scope.get_binding, @filename || __FILE__, @line || __LINE__)
```

Comment:
Is this really the same as `scope.output`?

Reply:
Erm, no. /o\